01_stratified_classification/nbconverted/plot_univariate_fs_results.py

```diff
@@ -0,0 +1,335 @@
+    index_names = list(rename_map.values())
+    rename_map[len(labels)] = 'id'
+    df = (pd.DataFrame(df_ix)
+        .rename(columns=rename_map)
+        .set_index(index_names)
+    )
+    # and return as series
+    return df['id']
+
+
+# In[11]:
+
+
+upset_series = series_from_samples(
+    list(fs_method_coefs.values()), list(fs_method_coefs.keys())
+)
+upset_series[upset_series != 0].sort_values(ascending=False).head(5)
+
+
+# In[12]:
+
+
+up.plot(upset_series[upset_series != 0])
+
+
+# In[13]:
+
+
+fs_method_small = {
+    k: v for k, v in fs_method_coefs.items() if 'mad_n1000' not in k
+}
+upset_series_small = series_from_samples(
+    list(fs_method_small.values()), list(fs_method_small.keys())
+)
+upset_series_small[upset_series_small != 0].sort_values(ascending=False).head(5)
+
+
+# In[14]:
+
+
+up.plot(upset_series_small[upset_series_small != 0])
+
+
+# We can see in the upset plots that the median and MAD methods (the ones that try to summarize/normalize across cancer types) tend to select the most distinct features relative to the other methods.
+# 
+# We can also see that the pan-cancer f-test and the median f-test tend to have a decent number of features in common, but not too many (much fewer than each of them alone). This is good in the sense that it means we're selecting fairly distinct features with the different methods.
+
+# ### Get distribution of univariate feature correlations
+# 
+# NOTE: these won't be exactly what was used for feature selection since we're not doing the same train/test splits here, or filtering cancer types by mutation count -- instead we're just calculating on the whole dataset. This could and probably should be fixed in the future, the actual distributions that we're selecting features based on could be quite different.
+
+# In[15]:
+
+
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import f_classif
+
+import pancancer_evaluation.utilities.data_utilities as du
+
+print('Loading gene label data...', file=sys.stderr)
+genes_df = du.load_top_50()
+sample_info_df = du.load_sample_info(verbose=True)
+
+# this returns a tuple of dataframes, unpack it below
+pancancer_data = du.load_pancancer_data(verbose=True)
+(sample_freeze_df,
+ mutation_df,
+ copy_loss_df,
+ copy_gain_df,
+ mut_burden_df) = pancancer_data
+
+rnaseq_df = du.load_expression_data(verbose=True)
+
+# standardize columns of expression dataframe
+print('Standardizing columns of expression data...', file=sys.stderr)
+rnaseq_df[rnaseq_df.columns] = StandardScaler().fit_transform(rnaseq_df[rnaseq_df.columns])
+
+
+# In[16]:
+
+
+print(rnaseq_df.shape)
+rnaseq_df.iloc[:5, :5]
+
+
+# In[17]:
+
+
+y_df = (mutation_df
+    .loc[:, [gene]]
+    .merge(sample_freeze_df, left_index=True, right_on='SAMPLE_BARCODE')
+    .drop(columns='PATIENT_BARCODE')
+    .set_index('SAMPLE_BARCODE')
+    .rename(columns={gene: 'status',
+                     'DISEASE': 'cancer_type',
+                     'SUBTYPE': 'subtype'})
```

Comment:
Is the subtype being used in the model? Because I imagine you don't have that info for all cancer types and it seems like that could really affect the model...

Reply:
Currently we're using it in the train/test set stratification - in other words, if a cancer type has subtypes, we try to balance the subtypes between train/test sets so that proportions are relatively equal. For cancer types that don't have well-defined subtypes we just ignore it and stratify by the cancer type as a whole.

We're not actually using it for prediction (i.e. as a feature in the model), though, for the reason you suggested.